include/net/dsa.h

```diff
@@ -0,0 +1,174 @@
+/** @file
+ @brief DSA definitions and handlers
+
+ */
+
+/*
+ * Copyright (c) 2020 DENX Software Engineering GmbH
+ *               Lukasz Majewski <lukma@denx.de>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_DSA_H_
+#define ZEPHYR_INCLUDE_NET_DSA_H_
+
+#include <device.h>
+#include <net/net_if.h>
+
+/**
+ * @brief LLDP definitions and helpers
+ * @defgroup lldp Link Layer Discovery Protocol definitions and helpers
+ * @ingroup networking
+ * @{
+ */
+
+#define NET_DSA_PORT_MAX_COUNT 8
+
+#define DSA_NET_DEVICE_INIT(dev_name, drv_name, init_fn, pm_control_fn,	\
+			    data, cfg_info, prio, api, mtu, port_num)	  \
+	DEVICE_DEFINE(dev_name, drv_name, init_fn, pm_control_fn, data,	\
+		      cfg_info, POST_KERNEL, prio, api);		\
+	NET_L2_DATA_INIT(dev_name, 0, NET_L2_GET_CTX_TYPE(ETHERNET_L2)); \
+	NET_IF_INIT(dev_name, 0, ETHERNET_L2, mtu, port_num)
```

Comment:
The last parameter of NET_IF_INIT() is used in VLAN. Overriding that value for other purposes means that one cannot use VLAN with DSA. Is this desired/ok?

Reply:
This define was a leftover from other (now old) approach to provide DSA. Instead of using DSA_NET_DEVICE_INIT() I've decided to use NET_DEVICE_INIT_INSTANCE for each slave port of DSA switch.